go/libraries/doltcore/sqle/dtables/diff_table.go

```diff
@@ -958,3 +958,74 @@ func CalculateDiffSchema(fromSch, toSch schema.Schema) (schema.Schema, error) {
 
 	return schema.UnkeyedSchemaFromCols(schema.NewColCollection(cols...)), nil
 }
+
+//------------------------------------
+// diffTableFunctionRowIter
+//------------------------------------
+
```

Comment:
It's a little odd to have the diffTableFunctionRowIter in the `diff_table.go` file, since it isn't used by diff_table. Putting this into `diff_iter.go` would probably be better organized. 

Also, might make more sense to rename it to something like `DiffPartitionRowIter` or something, since otherwise the `DiffTableFunction` name is a little misleading if we're using it for other purposes besides the `dolt_diff()` table function.